File: csharp/DVSClient/DVSClient/server/address/format/RestApiEnrichmentResultAddress.cs
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Format
{
    public class RestApiEnrichmentResultAddress
    {
        [JsonProperty("geocodes")]
        public RestApiEnrichmentDatasetGeocodes? Geocodes { get; set; }

        [JsonProperty("usa_regional_geocodes")]
        public RestApiEnrichmentUsaRegionalGeocodes? UsaRegionalGeocodes { get; set; }

        [JsonProperty("aus_regional_geocodes")]
        public RestApiEnrichmentAusRegionalGeocodes? AusRegionalGeocodes { get; set; }

        [JsonProperty("nz_regional_geocodes")]
        public RestApiEnrichmentNzlRegionalGeocodes? NzRegionalGeocodes { get; set; }

        [JsonProperty("uk_location_complete")]
        public RestApiEnrichmentUKLocationComplete? GbrLocationComplete { get; set; }

        [JsonProperty("uk_location_essential")]
        public RestApiEnrichmentUKLocationEssential? GbrLocationEssential { get; set; }

        [JsonProperty("gbr_government")]
        public RestApiEnrichmentGbrGovernment? GbrGovernment { get; set; }

        [JsonProperty("gbr_business")]
        public RestApiEnrichmentGbAdditional? GbrBusiness { get; set; }

        [JsonProperty("gbr_health")]
        public RestApiEnrichmentGbrHealth? GbrHealth { get; set; }

        [JsonProperty("premium_location_insight")]
        public RestApiPremiumLocationInsight? PremiumLocationInsight { get; set; }

        [JsonProperty("what3words")]
        public RestApiEnrichmentWhat3Words? What3words { get; set; }
    }
}

File: typescript/src/server/address/format/restApiEnrichmentResultAddress.ts
import { RestApiEnrichmentAusRegionalGeocodes } from "./restApiEnrichmentAusRegionalGeocodes";
import { RestApiEnrichmentDatasetGeocodes } from "./restApiEnrichmentDatasetGeocodes";
import { RestApiEnrichmentGbAdditional } from "./restApiEnrichmentGbAdditional";
import { RestApiEnrichmentGbrGovernment } from "./restApiEnrichmentGbrGovernment";
import { RestApiEnrichmentGbrHealth } from "./restApiEnrichmentGbrHealth";
import { RestApiEnrichmentNzlRegionalGeocodes } from "./restApiEnrichmentNzlRegionalGeocodes";
import { RestApiEnrichmentUKLocationComplete } from "./restApiEnrichmentUKLocationComplete";
import { RestApiEnrichmentUKLocationEssential } from "./restApiEnrichmentUKLocationEssential";
import { RestApiEnrichmentUsaRegionalGeocodes } from "./restApiEnrichmentUsaRegionalGeocodes";
import { RestApiEnrichmentWhat3Words } from "./restApiEnrichmentWhat3Words";
import { RestApiPremiumLocationInsight } from "./restApiPremiumLocationInsight";

export type RestApiEnrichmentResultAddress = {
    geocodes?: RestApiEnrichmentDatasetGeocodes;
    usa_regional_geocodes?: RestApiEnrichmentUsaRegionalGeocodes;
    aus_regional_geocodes?: RestApiEnrichmentAusRegionalGeocodes;
    nz_regional_geocodes?: RestApiEnrichmentNzlRegionalGeocodes;
    uk_location_complete?: RestApiEnrichmentUKLocationComplete;
    uk_location_essential?: RestApiEnrichmentUKLocationEssential;
    gbr_government?: RestApiEnrichmentGbrGovernment;
    gbr_business?: RestApiEnrichmentGbAdditional;
    gbr_health?: RestApiEnrichmentGbrHealth;
    premium_location_insight?: RestApiPremiumLocationInsight;
    what3words?: RestApiEnrichmentWhat3Words;
};